pkg/exchange/bybit/convert.go

```diff
@@ -169,15 +189,44 @@ func toGlobalOrderStatus(status bybitapi.OrderStatus) (types.OrderStatus, error)
 	}
 }
 
-func hashStringID(s string) uint64 {
-	h := fnv.New64a()
-	h.Write([]byte(s))
-	return h.Sum64()
-}
+// processQuantity converts the quantity unit from quote coin to base coin if the order is a **MARKET BUY**.
+//
+// If the status is OrderStatusPartiallyFilled, it returns the estimated quantity based on the base coin.
+//
+// If the order status is OrderStatusPartiallyFilledCanceled, it indicates that the order is not fully filled,
+// and the system has automatically canceled it. In this scenario, CumExecQty is considered equal to Qty.
+func processQuantity(o bybitapi.Order) (fixedpoint.Value, error) {
+	if o.Side != bybitapi.SideBuy || o.OrderType != bybitapi.OrderTypeMarket {
+		return o.Qty, nil
+	}
+
+	var qty fixedpoint.Value
+	switch o.OrderStatus {
+	case bybitapi.OrderStatusPartiallyFilled:
+		// if CumExecValue is zero, it indicates the caller is from the RESTFUL API.
+		// we can use AvgPrice to estimate quantity.
+		if o.CumExecValue.IsZero() {
+			qty = o.Qty.Div(o.AvgPrice)
```

Comment:
Just to double check that when there is no cumulative executed quantity, will `o.AvgPrice` be zero?
If it is not zero, what does `o.AvgPrice` mean?

Reply:
> Just to double check that when there is no cumulative executed quantity, will o.AvgPrice be zero?
yes.
> If it is not zero, what does o.AvgPrice mean?
AvgPrice: Average filled price. If unfilled, it is "".  

In our converter of fixedpoint.Value, it will be converted to zero if not filled.
You reminded me, i will add a sanity check to avoid a panic caused by AvgPrice being equal to 0.